client/src/components/ui/license-filter-dropdown.tsx

```diff
@@ -0,0 +1,63 @@
+import { ChangeEvent, useState } from 'react'
+import { ExpandMore } from '@mui/icons-material';
+
+
+interface DropdownOptionProps {
+  option: string
+  isPlaceholder?: boolean
+}
+
+interface LicenseFilterDropdownProps {
+  options: string[]
+  placeholderOption?: string 
+}
+
+
+const DropdownOption = ({option, isPlaceholder=false}: DropdownOptionProps) => {
+  return (
+    <option 
+      disabled={isPlaceholder}
+      hidden={isPlaceholder}
+      value={isPlaceholder ? "" : option}
+    >
+      {option}
+    </option>
+  )
+}
+
+const LicenseFilterDropdown = ({options, placeholderOption}: LicenseFilterDropdownProps) => {
+  const [selected, setSelected] = useState<string>(placeholderOption ? "" : options[0])
+  const fullOptions = placeholderOption ? [placeholderOption, ...options] : options
+  
+  const handleSelect = (e: ChangeEvent<HTMLSelectElement>) => {
+    setSelected(e.target.value)
+  }
+
+  console.log(selected)
+  return (
+    <div 
+      className={`filter-container relative inline-flex h-[40px] w-[160px] px-[16px] py-[8px] rounded-[8px] bg-[#F2F2F2]`}
+    >
+      <select
+        name="license-dropdown-filter" 
+        id="license-dropdown-filter"
+        value={selected}
+        className={`appearance-none bg-[#F2F2F2] w-full text-[14px]`}
+        onChange={handleSelect}
+      >
+        {fullOptions.map((opt, idx) =>  (
+          <DropdownOption 
+            key={idx} 
+            option={opt} 
+            isPlaceholder={idx === 0 && !!placeholderOption}
+          />
+        ))}
+      </select>
```

Comment:
🌴 🌴 🌴  Super non-blocking comment:
Consider separating the placeholder option from the other dropdown options. You'd be able to simplify the logic of this file.

e.g. something like:
```typescript
{placeholderOption && (
  <option disabled value="">
    {placeholderOption}
  </option>
)}
{options.map((opt, idx) => (
  <option key={opt} value={opt}>
    {opt}
  </option>
))}
```

That said, this is functional and readable, so no big deal 😇